### ISSUE TITLE
Keyboard and mouse get disabled during crouton install

### ISSUE THREAD

**Issue by TheStally** [dnschneid/crouton#1994]
Hi

I've recently power washed my chromebook and am looking to setup crouton again, which was previously running fine.

I downloaded the latest crouton and have been trying to install using the standard "-t xfce" to get ubuntu with xfce, however during the install process (around the time it's downloading/installing packages i think) my mouse and keyboard both stop working completely.

The install continues as far as entering the username, however by this point I cant do anything except hold the power button down and restart my chromebook.  Then the mouse/keyboard works fine.

I'm using an Acer Chromebook 11 (CB3-111), and have had this setup previously.

Anyone got any ideas?

**Comment by tjcravey** [comment]
Same problem here.  Samsung XE303C12

I tried it again... on the second pass the mouse stopped working, but the keyboard worked, so I was able to complete the install.

**Comment by Ephemerial** [comment]
I am experiencing a similar problem with a Toshiba Chromebook 2. Both the keyboard and touchpad stop responding to input part-way through the installation process. The issue is inconsistent and seems to occur roughly ~70% of the time, under the exact same circumstances.

**Comment by tista500** [comment]
Greetings,

Same issue +1 on my Asus C200 with trusty fresh intall.
In my case during installation:
* internal keyboard - dead
* internal clickpad - works
* external USB keyboards - dead
* external Bluetooth keyboard - works

Regards.

**Comment by danopia** [comment]
I posted about this on reddit: https://www.reddit.com/r/Crouton/comments/3i6axn/whenever_i_built_a_chroot_on_my_pixel_2015_it/

At one point running an install with a dead internal keyboard (from a previous chroot build) fixed the keyboard and broke the trackpad/touchscreen.

**Comment by chrisware** [comment]
Same thing happened to me installing crouton/trusty/unity (which I've done several times on my Pixel 2015 LS in the last few days). Either the trackpad stopped and/or keyboard and/or touchscreen. The install gets to the last step but I can't complete the last step and enter the user name. Workaround was to power off the chromebook, restart then either start a fresh install or enter the chroot and when prompted to finish the install press Y. The freezing happens whether I'm using the chromebook during install, or not touching it.

**Comment by sarbon15** [comment]
I am also experiencing this issue on my Acer C720 while trying to install trusty/unity. The first time through, the mouse was disabled, and then the keyboard. Then the second time, the keyboard and then the mouse.

**Comment by mkvalor** [comment]
Same issue: 2014 HP Chromebook 14 (Intel).  Deleted my old Utopic CLI chroot, deleted crouton script, downloaded latest crouton script tonight and ran with -r trusty -t kde-desktop,chrome,extension. (did not use -e)  The trackpad stopped working fairly early in the install (didn't try the keyboard at that point).  Some time later, during package updating, I noticed the trackpad worked again.   But when the script reached the point that it asked for a username, the keyboard would not respond.  All I could do was use the mouse to shut down.

**Comment by dreamwavedev** [comment]
Only real thing I can think of would be using the virtual keyboard to get through install...no idea what that could cause if the problem is with the actual driver post-installation

**Comment by danopia** [comment]
Attaching a USB keyboard/mouse after the bug takes affect seems to
consistently work for me as a workaround. Reattaching a bugged USB device
seems to recover it without a reboot as well.

**Comment by dnschneid** [comment]
Possibly due to restarting udev (just a guess though).  #2072 may fix this.

**Comment by fredrikaverpil** [comment]
I've been struggling with this issue today and finally went here to post about it. Duh, should have searched the issues beforehand. I'm posting my full report here in case that helps finding the root cause.

---

Every time I create a new chroot using crouton, the keyboard and/or touchpad on my Dell Chromebook 11" becomes completely unresponsive (and cursor disappears) during the installation process. I would initially dismiss this as a hardware issue, but bare with me...

- A simple logout in Chrome OS fixes the issue and makes both keyboard and touchpad responsive again.
- The keyboard and touchpad *never* becomes unresponsive while using Chrome OS or while using Ubuntu running next to Ubuntu.
- When experiencing unresponsiveness from the touchpad and when plugging in a USB mouse, the cursor reappears and I can use the mouse while the touchpad remains unresponsive.

I've disabled ALL Chrome extensions and I still experience this everytime during crouton's Linux installation. I've performed a powerwash. I've changed the channel from stable to beta. I've changed back the beta onto stable channel. I've executed and installed crouton onto an ext4 formatted SD card. I've opened up the casing and disconnected the battery completely. I'm running out of things to try to avoid this issue.

It *always* and *only* happens during crouton execution. Strange, yes?

Example command:

    sudo sh -e ~/Downloads/crouton -t core,audio,chrome,unity -r trusty -n trusty-unity

Obviously, I've enabled developer mode on my Chromebook.

    Version 45.0.2454.86 (64-bit)
    Platform 7262.52.0 (Official Build) stable-channel wolf
    Firmware Google_Wolf.4389.24.62

tl;dr As long as I don't use crouton, I've got no issues. But when I use crouton, my keyboard and/or my touchpad *always* stops working completely.

**Comment by allenp** [comment]
Plugging in an external keyboard allowed me to finish the install, then I pressed power button to go to logon screen and used tab to get to the sign out button. From there I signed out (which pushed me to the shell screen, so I typed exit to complete the logout).

From there my trackpad and internal keyboard started working again.

**Comment by dnschneid** [comment]
Pretty certain it's restarting udev.  Filed a [crbug](https://code.google.com/p/chromium/issues/detail?id=539742).  #2072 should fix the issue, although I haven't tested it yet.

**Comment by dnschneid** [comment]
Huh, #2072 doesn't actually fix this.  More digging required...

**Comment by brmbrmcar** [comment]
Happened to me to.

**Comment by hotpocket** [comment]
+1 new chromebook pixel 2015 and was hard power cycling the chromebook to get it to respond, resuming the install with -u ... my experience with this install is night & day with prior efforts which were a breeze by comparison.

**Comment by nwhusted** [comment]
Having the same problem with an Acer CB5-571 running ChromeOS 45.0.2454.98. The keyboard and mouse issue did not start until this or the immediately previous ChromeOS version.

**Comment by emanb29** [comment]
Confirmed on HP Chromebook 14 (cr3114 HPSA) with all Ubuntu installations

**Comment by ds300** [comment]
This is happening for me too, Pixel 2 ChromeOS 46.0.2490.71

**Comment by jimt** [comment]
FWIW, while I did need a USB keyboard when installing the xfce target, the cli-extra target went without hitch (new Toshiba Chromebook 2).

**Comment by rkhayat** [comment]
Just happened to me on Pixel LS 2015!

**Comment by ProfessorKaos64** [comment]
Same here on Acer C720. Joining in on "the fun" :)

**Comment by dougfelt** [comment]
Me too, Acer 4Mb/128Mb.  First install of ubuntu precise (in August) went fine.  Just tried to install trusty, keyboard and mouse became unresponsive shortly before it asked for the default user name.

**Comment by leewc** [comment]
Just to leave this as for others if they don't have a USB Keyboard but a USB mouse only, you can enable the virtual keyboard beforehand under Settings > Accessibility > Enable On-Screen Keyboard

After which install your chroot (without the USB mouse plugged in), and once it get's stuck, stick it in. If it doesn't work then reinsert it, took me a few tries to get the mouse responsive.

Then perform the username password set up, with the mouse, once complete do a reboot.

**Comment by salil-bhargava** [comment]
I have a Toshiba Chromebook with Ubuntu, so its not really Crouton specific, but the issue was teh Synaptics trackpad package. 

sudo apt-get install xserver-xorg-input-synaptics

**Comment by darcyparker** [comment]
@leewc - thanks for that workaround of enabling the on screen keyboard. It's a pain, but for me easier than getting a usb keyboard right now. It worked.

**Comment by leewc** [comment]
@darcyparker glad to have helped! I was in the same issue as you and only had a wireless mouse with me! Hmm, Perhaps I should add this to the docs.

**Comment by dnastase** [comment]
Encountered the same bug: Acer Chromebook R11

**Comment by wxv** [comment]
I have the same issue on Acer C740. It's mouse not working, keyboard not working, or both. Logging out fixes the issue, but stops the installer.

**Comment by brismuth** [comment]
This is happening to me on an Acer R11 Chromebook during Ubuntu installation. I got around it using the usb mouse as suggested, but the keyboard still becomes deactivated sometimes when I boot up the chroot. It is usually fine, but it has deactivated the keyboard twice now during chroot boot.

**Comment by rumpaar1** [comment]
Same issue on an HP Chromebook 14 with Intel Processor. Onscreen keyboard works fine in Chrome OS and hopefully will be able to use it to finish the install.

**Comment by rumpaar1** [comment]
Wasn't able to test on-screen keyboard for finishing Crouton install (install failed as I ran out of space on my SD card, but worked everywhere in Chrome OS I tried). Am trying again, this time both keyboard and trackpad stopped working. However, plugging in an external keyboard and mouse worked. Waiting on install to finish.

**Comment by divx118** [comment]
@rumpaar1 You can also do a force reboot when you need to enter your username then use `sudo enter-chroot` on the restart and complete the install script. However what you did does also work fine.

**Comment by camitch** [comment]
Experiencing this issue on original HP 14 cr3114 "Falco" 

Update to latest version of Chrome OS stable.  Newest version of crouton script

Install precise - no issue
Install trusty - mouse and keyboard function die during install (possibly finished? couldn't click over to crosh window) 
Install jessie - everything works fine until log out from jessie (running Xfce) touchpad dies

Did attempt to install trusty twice, no issues on second install attempt

**Comment by InsanityOnABun** [comment]
Same issue on Acer C910. Force restart and sudo enter-chroot to start the install script again worked as a workaround.

**Comment by Just-Johnny** [comment]
Similar issue on my Samsung Chromebook 2. However, I only lose keyboard support and it only happens when installing Trusty. The default Precise works fine.

**Comment by fengye** [comment]
Same here, Toshiba Chromebook 2. Once has disabled both touchpad and keyboard. The second time only the keyboard get disabled. Using touch pad to logout and connect a USB keyboard solve the problem.

**Comment by CJEdgerton** [comment]
@fengye I had the same issue on my TCB2. I used a USB keyboard to set the username and pwd at the end of the install, and then rebooted. After reboot everything was working again.

**Comment by spovich** [comment]
I'm experiencing the same problem with Trusty on a Dell Chromebook 13" 7310.  Attaching external USB keyboard allowed me to finish install.  Precise did not have the problem.

**Comment by idvoretskyi** [comment]
I have the same issue.

Possible workaround - to use screen keyboard on my Chromebook Pixel, but it would be great to find a solution.

**Comment by danopia** [comment]
@idvoretskyi That only works until you build another chroot and the touchscreen disables too :) but there's a chance of the keyboard coming back when that happens.

This ticket has been open for a while with no visible progress. Would a bounty help?

**Comment by cvarga15** [comment]
I'm getting the same issue on my Dell Chromebook 11 (CANDY). Ctrl + Alt + T doesn't bring up crosh shell anymore when I'm in ChromeOS even after reboot.

**Comment by legostax** [comment]
As a suggestion to the dev, why not set up crouton to prompt for username and password at the beginning of the install, then at the end, reboot?  This could be set with an optional argument.  Then users wouldn't have to fiddle with a USB keyboard or mouse when finishing the install/setup.

**Comment by Salvyman** [comment]
I know this isn't a solution to WHY the keyboard and mouse pad can be disabled when installing crouton, however there is a workaround to where you don't have to reboot your chromebook and hope that the process doesn't repeat - use the built-in on screen keyboard. You can activate it by clicking on your profile picture on the bottom left of the screen, click on advanced settings, scroll down to accessibility and check the Onscreen keyboard tab. A keyboard icon will appear on tabs menu on the left, which you can click on. Use it to type in your account name and password. And voila! Restart your computer and your ready to go

**Comment by brmbrmcar** [comment]
But that won't help without an external mouse because the touchpad is unusable.

**Comment by Salvyman** [comment]
Oh right, duh. Haha, i feel dumb. I actually have a wireless mouse that I used, which actually wasn't working during the chroot installation. However, unplugging it and plugging it back in fixed it

**Comment by legostax** [comment]
Yeah, you'd still need an external mouse.  That's what I do.  I use the onscreen keyboard with my bluetooth mouse.
Perhaps after the install is complete, crouton could also trigger a reboot so users wouldn't be stuck trying to reboot without their mouse and keyboard.

**Comment by brmbrmcar** [comment]
I just attach both an external mouse and keyboard.

**Comment by zxvv** [comment]
Another workaround for this is to run crouton from the chromebook console.

 - Before running crouton, hit ctrl-alt-"->" above the 3 key to switch to a console VT2 (virtual terminal 2).
 - Login as chronos
 - run crouton...

one cannot use cut-and-paste in the VT, which is less convenient, but it does prevent the lockup.

**Comment by drinkcat** [comment]
Sorry for the unbelievably long delay in investigating this...

This is due to package `xserver-xorg-input-vmmouse`, that runs this in its post-install hook:
```
udevadm trigger --subsystem-match=input --action=change
```
Running this locks the input on Chrome OS side (whether inside or outside the chroot, more than 1 attempt may be required).

Let's not install `xserver-xorg-input-vmmouse` as a workaround, then let's try to figure out what's wrong in Chromium OS.

**Comment by cvarga15** [comment]
Thanks guys!!

**Comment by zxvv** [comment]
Awesome!

I've tried to gather some info...

Looks like the udev rule for vmmouse change invokes /bin/vmmouse_detect.

http://bazaar.launchpad.net/~ubuntu-branches/ubuntu/wily/xserver-xorg-input-vmmouse/wily/view/head:/tools/69-xorg-vmmouse.rules.in

http://bazaar.launchpad.net/~ubuntu-branches/ubuntu/wily/xserver-xorg-input-vmmouse/wily/view/head:/tools/hal-probe-vmmouse.in

So I tried invoking /bin/vmmouse_detect inside a debian stretch chroot. -- and it did not lock up the console (it had no readily apparent effect at all).

So maybe it's one of the other rules that is causing the lockup.
I dumped all the udev rules on my pixel, and it looks like there are 22 rules for 'input' subsystem.
I'll can continue trying some of these to see whether there's a specific rule that causes the lockup.

The link below is the output of 'udevadm info -e' on a chromebook pixel, filtered for only those entries with SUBSYSTEM=input'.  I think this is 22 separate rules.

[udevadm-info-output-for-input-subsystem.txt](https://github.com/dnschneid/crouton/files/230940/udevadm-info-output-for-input-subsystem.txt)

**Comment by micber** [comment]
I can do a full of trusty but track pad is unresponsive but keyboard works during install.  Acer 13 new. Guess I will try precise

**Comment by radix** [comment]
This is still happening for me on a Pixel 2. Just downloaded a fresh crouton, and tried to set up a new xenial chroot with xfce and xiwi targets. I stepped away and when I came back to see the "Please specify a username for the primary user:" prompt, I can't type or use my mouse.

**Comment by DennisLfromGA** [comment]
*Wow, this must be a real stumper!*

Two PRs ( https://github.com/dnschneid/crouton/pull/2072 & https://github.com/dnschneid/crouton/pull/2541  ) have been issued and merged and two crbugs ( [#539742](https://bugs.chromium.org/p/chromium/issues/detail?id=539742) & [#605011](https://bugs.chromium.org/p/chromium/issues/detail?id=605011) ) have been filed and still no joy.

> *Arguably, one possible solution is also "Do not do that ,-)".*

And even more arguably, that is **not** an option for us crouton users. :(

I plan to install 'xenial' soon so I'll be sure to report back with my experience.

-**DennisL**

**Comment by DennisLfromGA** [comment]
Reporting back:

I used @zxvv's method above (https://github.com/dnschneid/crouton/issues/1994#issuecomment-212232028) by logging into to my account and then opening VT2 to do the installation. The keyboard and mouse continue to  work great with no interruptions. Until this issue is good and fixed, I'll continue to use this method.

-----
I installed 3 chroots (using @dnschneid's *mergeboat* branch courtesy of @zxvv 👍)
- **xenial/lxqt,mate** (worked great but the lxqt & mate packages together led to a jumbled mess)
- **xenial/lxqt** (works great but is pretty vanilla)
- **xenial/mate** (works great and with a few additional packages is my favorite)

The mate target isn't quite ready for prime-time but I got it installed by just looking at the 'mate' target file and installing some of it manually.

I've been able to run all of the above in both 'xorg' and 'xiwi' with no problems.

-**DennisL**

**Comment by jasonpettus** [comment]
Just wanted to mention that this is still happening -- I'm on an ASUS C300 and am having my keyboard and/or trackpad lock up during installs. Interestingly, it doesn't happen every time, nor is it both every time -- sometimes my trackpad works fine and sometimes it's completely disabled, and sometimes I'll do an install and have no problems at all.

**Comment by 3rdwiki** [comment]
@dnschneid I am interested to know how you would de-bug this at startup and during the install process. What do I need to launch ?
Any links would be v. much appreciated.

**Comment by floridawilson** [comment]
Glad to visit this site. I found the information which I was looking for. Recently I bought a wireless mouse and keyboard and I connected to my Dell laptop after that it shows the connection error and i took help from [Dell laptop support](https://www.delltechsupportnumbers.com/dell-laptop-support/) and they helped me lot to fix this problem.
